src/main/java/recipeapplication/controllers/RecipeController.java

```diff
@@ -38,4 +34,21 @@ private Recipe toEntity(RecipeTO recipe) {
 
 		return recipeEntity;
 	}
+
+	@RequestMapping(value = "/putRecipeById", method = RequestMethod.PUT)
+	public void recipePut(@RequestParam String id,
+						   @RequestPart(value = "recipeImage", required = false) MultipartFile file,
+						  @ModelAttribute("recipe") RecipeTO recipe) {
+		if (id.equalsIgnoreCase(recipe.getId()) &&
+				recipe.getCategory() != null &&
+				recipe.getIngridiant() != null &&
+				recipe.getPreperation() != null)
+			this.recipes.updateRecipe(toEntity(recipe));
+		}
+
+	@RequestMapping(value = "/requestRecipeById", method = RequestMethod.GET)
+	public Recipe recipeGet(@RequestParam String id)
```

Comment:
@RequestParam("id")

Reply:
Changed to use @PathVariable("id") instead of @RequestParam